### ISSUE TITLE
Nomad, Consul, and Vault integration question (Consul Secret Engine)

### ISSUE THREAD

**Issue by idrennanvmware** [hashicorp/nomad#8943]
This is really an integration question for the Nomad team around Vault and Consul.

If I am using Consul Secrets engine in Vault, and then using the token issued by the consul secrets engine as an input in to my job using the "-consul-token" flag - as I understand it that token is stored internally to Nomad along with my job.   Now, when that token expires, how does Nomad know (I assume the job just starts to fail to schedule since it no longer can access the Consul KV entry specified in the job file) the token now needs to be renewed?   Is this something that can be automated, or do we have to have a process that periodically redeploys the job with the new token PRIOR to the old one expiring?

tldr:  whats the recommended way of rotating -consul-token values, for a job, when the token expires?

Thanks!

**Comment by schmichael** [comment]
The [Job's Consul Token](https://pkg.go.dev/github.com/hashicorp/nomad/nomad/structs#Job.ConsulToken) behaves similar to the [Job's Vault Token](https://pkg.go.dev/github.com/hashicorp/nomad/nomad/structs#Job.VaultToken): both are used for checking permissions and neither are persisted.

This means the token (whether Consul or Vault) only needs to valid at job-submit-time. Once Nomad confirms the token has permission to perform the desired operations (eg permission to register a Consul service or access a Vault role), the token is stripped from the Job and new tokens are created on a per-allocation basis.

So there's no need to refresh expired Consul or Vault tokens used to register a job! They're only used at submit-time.
